tuf/api/metadata.py

```diff
@@ -114,6 +114,36 @@ def __init__(self, signed: T, signatures: "OrderedDict[str, Signature]"):
         self.signed: T = signed
         self.signatures = signatures
 
+    def __eq__(self, other: "Any") -> bool:
```

Comment:
```suggestion
    def __eq__(self, other: Any) -> bool:
```
Same comment applies elsewhere in this PR.